server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java

```diff
@@ -274,23 +298,62 @@ public String executor() {
         }
     }
 
-    TransportResponse handleExtensionRequest(ExtensionRequest extensionRequest) throws Exception {
-        // Read enum
-        if (extensionRequest.getRequestType() == RequestType.REQUEST_EXTENSION_CLUSTER_STATE) {
-            ClusterStateResponse clusterStateResponse = new ClusterStateResponse(
-                clusterService.getClusterName(),
-                clusterService.state(),
-                false
+    /**
+     * Handles a {@link RegisterRestActionsRequest}.
+     *
+     * @param restActionsRequest  The request to handle.
+     * @return  A {@link RegisterRestActionsResponse} indicating success.
+     * @throws Exception if the request is not handled properly.
+     */
+    TransportResponse handleRegisterRestActionsRequest(RegisterRestActionsRequest restActionsRequest) throws Exception {
```

Comment:
Just wrapping my head around this handler. Can we move this to a new file and create a new case in `handleExtensionRequest`? This is to keep a common handler for all the requests coming from Extensions.

Reply:
I'll consider that as part of the next PR.  Right now we're just handling a list of Strings but it will change to handling Actions with some more complexities involved.